tools/flight_recorder/components/utils.py

```diff
@@ -5,22 +5,62 @@
 # LICENSE file in the root directory of this source tree.
 
 import argparse
+import logging
 import math
-from typing import Any, Dict, List, Set, Tuple
+from typing import Any, Callable, Dict, List, Optional, Set, Tuple
 
-from tools.flight_recorder.components.types import (
-    Group,
-    MatchState,
-    Membership,
-    Op,
-    P2P,
-)
+from .types import Group, MatchState, Membership, Op, P2P
```

Comment:
nit: not necessary to be in this PR, but you can move it to a different file named "logger"? up to you